src/bench/wallet_loading.cpp

```diff
@@ -46,10 +46,10 @@ static void BenchUnloadWallet(std::shared_ptr<CWallet>&& wallet)
 static void AddTx(CWallet& wallet)
 {
     const auto& dest = wallet.GetNewDestination(OutputType::BECH32, "");
-    assert(dest.HasRes());
+    assert(dest);
```

Comment:
This pull changes both the interface, as well as the internal representation. I wonder if changing the interface and renames can be done separate from the restructuring.

Hopefully those would be uncontroversial and could be reviewed easier/faster, so that less silent merge conflicts arise as the (class gets used more and more.

Reply:
> This pull changes both the interface, as well as the internal representation. I wonder if changing the interface and renames can be done separate from the restructuring.

Only the third commit drops BResult usages, so the answer should be yes. IMO std::optional has a nice interface and it is a shame to reinvent it with stranger HasRes GetObj ReleaseObj methods.

Reply:
With separate I meant separate pulls :sweat_smile:

Reply:
> With separate I meant separate pulls sweat_smile

Oh sorry, I thought you were asking literally if it is possible

Reply:
I think the stuff that can be split out:

* Makes Result class provide same operators and methods as std::optional
* Puts src/util/ code in the util:: namespace so naming reflects code organization and it is obvious where the class is coming from. Drops "B" from name because it is undocumented what it stands for (bilingual?)
* Supports Result<bilingual_str> return values. Drops ambiguous and potentially misleading BResult constructor that treats any bilingual string as an error, and any other type as a non-error. Adds util::Error constructor so errors have to be explicitly constructed and not any bilingual_str will be turned into one.

Reply:
I think these things are already split out. The first two commits do not change the interface or naming or functionality of BResult in any way. If you want me to move the last commit to another, PR I can do that.

Reply:
Yeah, I think that makes sense, so that we can first figure out the interface, and then start using it more widely, not the other way round.